tests/integration/test_grakn.py

```diff
@@ -217,6 +217,17 @@ def test_no_metatype_duplicates(self):
         id_set = set(concepts)
         self.assertEqual(len(id_set), 2) # entity and person, not the same
 
+    def test_query_errors_async(self):
+        answer = self.tx.query("match $x isa unicorn; get;")
+
+        with self.assertRaises(GraknError):
+            answer.get()
+
+    def test_query_errors_async_on_commit(self):
+        self.tx.query("match $x isa unicorn; get;")
+
+        with self.assertRaises(GraknError):
+            self.tx.commit()
```

Comment:
some tests on `.get()` blocking would be nice i think, if you can think of a way to do it :)

Reply:
I can't, not reliably in the current instrumentation anyway. I think we really need to revisit this at some later point as a client architecture revamp because there currently isn't much purpose in blocking.